ctru-rs/src/error.rs

```diff
@@ -1,28 +1,40 @@
 use std::error;
 use std::fmt;
 
+use ctru_sys::result::{R_DESCRIPTION, R_LEVEL, R_MODULE, R_SUMMARY};
+
 pub type Result<T> = ::std::result::Result<T, Error>;
 
 /// The error type returned by all libctru functions.
+#[non_exhaustive]
 pub enum Error {
-    Os(i32),
+    Os(ctru_sys::Result),
 }
 
-impl From<i32> for Error {
-    fn from(err: i32) -> Self {
+impl From<ctru_sys::Result> for Error {
+    fn from(err: ctru_sys::Result) -> Self {
         Error::Os(err)
     }
 }
 
 impl fmt::Debug for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            Error::Os(err) => write!(f, "libctru result code: {:08X}", err),
+            Error::Os(err) => f
+                .debug_struct("Error")
+                .field("raw", &format_args!("{:#08X}", err))
```

Comment:
nit: maybe prepend `0x` so it's clear this is a hex number?

Reply:
The `#` "alternate" marker does that automatically.